### PR TITLE
fix for relsub arm

### LINKED ISSUE #6068: asm.relsub in arm
asm.relsub should work in ARM:

Current output

```
         0x000081b0      0c009fe5       ldr r0, [0x000081c4]        ; [0x81c4:4]=0x81cc sym.main
         0x000081b4      0c309fe5       ldr r3, [0x000081c8]        ; [0x81c8:4]=0x8094 sym._init
```

What would be much better:

```
         0x000081b0      0c009fe5       ldr r0, sym.main
         0x000081b4      0c309fe5       ldr r3, sym._init
```

### PR DESCRIPTION
**Pull request by Svenito**
FIX #6068

### CODE REVIEW

File: libr/core/disasm.c [hunk 1]
```diff
@@ -731,9 +731,17 @@ static void ds_build_op_str(RDisasmState *ds) {
 					core->parser->flagspace = -1;
 				}
 			}
+			RCore *core = ds->core;
+			ut64 p = ds->analop.ptr;
+			char *msg = calloc (sizeof (char), 128); // is this big enough?
```
Comment (radare) [inline libr/core/disasm.c]: check return of calloc

File: libr/core/disasm.c [hunk 2]
```diff
@@ -731,9 +731,17 @@ static void ds_build_op_str(RDisasmState *ds) {
...
+			r_io_read_at (core->io, p, (ut8 *)msg, 128 - 1);
```
Comment (radare) [inline libr/core/disasm.c]: a pointer cant be bigger than 8 bytes (64bit)

File: libr/core/disasm.c [hunk 3]
```diff
@@ -731,9 +731,17 @@ static void ds_build_op_str(RDisasmState *ds) {
...
+			ut64 num = r_read_ble (msg, core->print->big_endian, ds->analop.refptr * 8);
```
Comment (radare) [inline libr/core/disasm.c]: We should check if ds->analop.refptr have a valid value before multiplying it (plugins can put a negative value or a huge one producing undesired effects

File: libr/core/disasm.c [hunk 4]
```diff
@@ -731,9 +731,17 @@ static void ds_build_op_str(RDisasmState *ds) {
...
+			RFlagItem *flag = r_flag_get_i (core->flags, num);
```
Comment (radare) [inline libr/core/disasm.c]: flag is not used :?

File: libr/core/disasm.c (hunk 2 above)
Comment (radare) [inline libr/core/disasm.c]: dat -1 looks wrong to me

Comment (radare) [inline libr/core/disasm.c]: you can use r_io_read_i() that allows to read a value of size from 1 to 8. without having to calloc, etc

cc @condret is this function also available in siol?

Comment (Svenito) [inline libr/core/disasm.c]: `r_io_read_at` takes a `ut8` pointer as its argument

**Comment by radare** [comment]
Use r_io_read_i() and remove all that calloc stuff

**Pull request merged**